server/test/test_filter.py

```diff
@@ -56,11 +56,11 @@ def test_complex_filter(self):
         filter_dict = ImmutableMultiDict(
```

Comment:
Suggest adding a more robust set of filter tests as they can get complex.   Example:
* categorical filters where the same value is specified multiple times, eg, `['foo', 'bar', 'foo']`
* all the range cases, eg, `*,1`, `1,*`, `*,*`, plus the illegals, eg, `100,0`, etc,

Reply:
👍